server/data/incidentTypeConfiguration/formWizard.ts

```diff
@@ -46,9 +48,168 @@ export function generateSteps(config: IncidentTypeConfiguration): FormWizard.Ste
       }
     })
 
+  groupSteps(steps)
+
   return steps
 }
 
+/**
+ * Merge all contiguous, non-branching steps in steps
+ *
+ * - If a step always lead to the same next step these can be potentially be
+ *   grouped.
+ * - Each step/question can only be in a single group.
+ *
+ * So steps with multiple parents (the right side of the branching)
+ * can't be the last step in a group. However these can be the
+ * beginning a new group.
+ *
+ * Example
+ * - Q-animals
+ *   => Q-dog
+ *   => Q-icecream
+ * - Q-dog
+ *   => Q-icecream
+ * - Q-icecream
+ *  => Q-end
+ * - Q-end
+ *  => null
+ *
+ * Here Q-animals branches and can lead to Q-dog or Q-icecream.
+ * This means q-animals would be in its own group.
+ * Both Q-animals and Q-dog lead to Q-icecream, so Q-icecream can't
+ * be part of these groups (each question belongs to a single group)
+ * Finally, Q-icecream always lead to Q-end, and Q-icecream is the only
+ * parent Q-end has, this means Q-end can be grouped into q-icecream
+ *
+ * @param steps to group
+ */
+function groupSteps(steps: FormWizard.Steps) {
+  const answersCounts: Map<string, number> = new Map()
+  const stepsWithSingleParent: Set<string | null> = buildStepsWithSingleParent()
+
+  /**
+   * Get the current number of answers for a given stepId
+   */
+  function getAnswersCountForStep(stepId: string) {
+    if (answersCounts.get(stepId) === undefined) {
```

Comment:
how about `answersCounts.has(stepId)`? did that cause one of the many strange js troubles?

Reply:
to be honest I had to fight with the JS collections so much that I almost gave up at some point...I think `has()` is the proper way to check if a Set has the element but I wouldn't be surprised if does something weird (I'm going to try, tests should either pass or fail, if they pass I'll replace 👌 )